### ISSUE TITLE
Memoization pattern improvement

### ISSUE THREAD

**Issue by jherax** [caiogondim/js-patterns-sublime-snippets#29]
When cached key stores a falsy value, it will always run the calculation.

I changed the line 10 in memoization snippet:
```
		// checks whether the key property was cached previously
		if (!funcMemoized.cache.hasOwnProperty(cacheKey)) {
```

**Comment by jherax** [comment]
Also works with: 

`if (!(cacheKey in funcMemoized.cache))`

**Comment by caiogondim** [comment]
Fixed on 9a5929ee7111fd7fea0a3be6f62af7adbe1cce68
Thanks a lot.
